### PR TITLE
Label sharding support

### PR DESCRIPTION
**Pull request by semion-akimtsev**
Enables label sharding support according to feature definition in https://github.com/SDVentures/Contour/issues/27

### CODE REVIEW

File: Sources/Contour/ConnectionPool.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@
     internal abstract class ConnectionPool<TConnection> : IConnectionPool<TConnection> where TConnection : class, IConnection
     {
         private readonly object syncRoot = new object();
-        private readonly ILog logger = LogManager.GetLogger<ConnectionPool<TConnection>>();
+        private readonly ILog logger = LogManager.GetLogger("ConnectionPool");
```
Comment (mzabolotko) [inline Sources/Contour/ConnectionPool.cs]: The logger name should contain a namespace.

File: Sources/Contour/Transport/RabbitMQ/Internal/Listener.cs [hunk 2]
```diff
@@ -109,6 +109,9 @@ public Listener(IBusContext busContext, IRabbitConnection connection, ISubscript
             this.validatorRegistry = validatorRegistry;
 
             this.ReceiverOptions = receiverOptions;
+            this.BrokerUrl = connection.ConnectionString;
+            this.logger = LogManager.GetLogger($"Listener(URL={this.BrokerUrl})");
```
Comment (mzabolotko) [inline Sources/Contour/Transport/RabbitMQ/Internal/Listener.cs]: The logger name should contain a namespace.

File: Sources/Contour/ConnectionPool.cs (hunk 1 above)
Comment (semion-akimtsev) [reply Sources/Contour/ConnectionPool.cs]: Please see the latest updates in the pull request

File: Sources/Contour/Transport/RabbitMQ/Internal/Listener.cs (hunk 2 above)
Comment (semion-akimtsev) [reply Sources/Contour/Transport/RabbitMQ/Internal/Listener.cs]: Please see the latest updates in the pull request

**Pull request merged**